common/test/common_tests/logic/swap_and_reset_test.cljc

```diff
@@ -0,0 +1,175 @@
+                                      0
+                                      nil
+                                      {})
+
+        file' (thf/apply-changes file changes)
+
+        ;; Get 
+        swapped (thf/get-shape-by-id file' (:id new-shape))]
+
+    ;; Check
+    (t/is (not= (:component-id component-1-copy-root) (:component-id swapped)))
+    (t/is (= (:id component-2) (:component-id swapped)))
+    (t/is (= (:id file) (:component-file swapped)))))
+
+(t/deftest test-swap-anidated
+  (let [;; Setup
+        file
+        (-> (thf/sample-file :file1)
+            (thc/add-simple-component :component-1 :component-1-main-root :component-1-main-child)
+            (thc/add-frame :component-container)
+            (thf/instantiate-component :component-1 :component-1-copy-root {:parent-label :component-container})
+            (thf/make-component :component-container-main :component-container)
+            (thf/instantiate-component :component-container-main :component-container-instance)
+            (thc/add-simple-component :component-2 :component-2-main-root :component-2-main-child))
+
+        page        (thf/current-page file)
+        component-1 (thf/get-component file :component-1)
+        component-2 (thf/get-component file :component-2)
+        component-3 (thf/get-component file :component-3)
+
+        copy
+        (->>
+         (thf/get-shape file :component-container-instance)
+         :shapes
+         first
+         (thf/get-shape-by-id file))
+
+        libraries {(:id  file) file}
+
+        ;; Action
+        [new-shape all-parents changes]
+        (cflh/generate-component-swap (pcb/empty-changes)
+                                      (:objects page)
+                                      copy
+                                      (:data file)
+                                      page
+                                      libraries
+                                      (:id component-2)
+                                      0
+                                      nil
+                                      {})
+
+        file'      (thf/apply-changes file changes)
+        libraries' {(:id  file') file'}
+        page'      (thf/current-page file')
+
+        ;; Get
+        swapped               (thf/get-shape-by-id file' (:id new-shape))
+        component-1-copy-root (thf/get-shape file' :component-1-copy-root)
+        slot                  (-> (ctf/find-swap-slot swapped
+                                                      page'
+                                                      file'
+                                                      libraries')
+                                  (ctk/build-swap-slot-group))]
+
+    ;; Check
+    (t/is (not= (:component-id copy) (:component-id swapped)))
+    (t/is (= (:id component-2) (:component-id swapped)))
+    (t/is (= (:id file) (:component-file swapped)))
+    (t/is (contains? (:touched swapped) slot))))
```

Comment:
I think you should also check that the slot points to the correct id. Something like:

`(t/is (= (ctk/get-swap-slot swapped') (:id copy))`